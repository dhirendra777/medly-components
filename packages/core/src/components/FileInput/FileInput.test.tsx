import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { FileInput } from './FileInput';

describe('FileInput component', () => {
    const foo = new File(['foo'], 'foo.txt', {
            type: 'text/plain'
        }),
        bar = new File(['bar'], 'bar.txt', {
            type: 'text/plain'
        });

    afterEach(cleanup);

    it('should render properly with default props', () => {
        const mockOnChange = jest.fn();
        // @ts-ignore
        const { container } = render(<FileInput files={[]} onChange={mockOnChange} />);
        expect(container).toMatchSnapshot();
    });

    it('should render properly with all props', () => {
        const mockOnChange = jest.fn();
        const { container } = render(
            <FileInput
                // @ts-ignore
                files={[foo, bar]}
                onChange={mockOnChange}
                fullWidth
                label="Email Address"
                labelPosition="top"
                required
                disabled
                description="We will never share your email with anyone"
                descriptionColor="grey"
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should call onFocus prop when click on the label', () => {
        const mockOnFocus = jest.fn();
        const mockOnChange = jest.fn();
        // @ts-ignore
        const { container } = render(<FileInput disabled files={[]} onFocus={mockOnFocus} onChange={mockOnChange} />);
        const label = container.querySelector('#medly-file-input-label');
        fireEvent.click(label);
        expect(mockOnFocus).toHaveBeenCalled();
    });

    it('should call onChange prop with expected files', async () => {
        const mockOnFocus = jest.fn();
        const mockOnChange = jest.fn();
        // @ts-ignore
        const { container } = render(<FileInput files={[]} onFocus={mockOnFocus} onChange={mockOnChange} />),
            input = container.querySelector('#medly-file-input');

        Object.defineProperty(input, 'files', {
            value: [foo, bar]
        });
        fireEvent.change(input);
        expect(mockOnChange).toBeCalledWith([foo, bar]);
    });
});
