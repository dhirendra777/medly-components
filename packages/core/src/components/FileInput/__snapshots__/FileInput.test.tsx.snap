// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput component should render properly with all props 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 8px 0px;
  cursor: default;
}

.c1::after {
  color: red;
  content: ' *';
}

.c5 {
  font-size: 14px;
  font-weight: 400;
  color: grey;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  grid-area: description;
  margin: 5px 0px 0px 2px;
  color: grey;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-areas: 'label' 'field' 'description';
}

.c0 .c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  width: 1px;
  opacity: 0;
}

.c3 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  background-clip: padding-box;
  background-color: #E0E0E0;
  box-sizing: border-box;
  border: 1px solid #E0E0E0;
  border-radius: 5px;
  cursor: not-allowed;
  text-align: center;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: calc(100% - 0px);
}

.c3:focus {
  border-color: #012040;
  outline: 0;
}

<div>
  <div
    class="c0"
    id="medly-file-input-field"
  >
    <label
      class="c1 c2"
      required=""
    >
      Email Address
    </label>
    <label
      class="c3"
      disabled=""
      id="medly-file-input-label"
    >
      foo.txt, bar.txt
      <input
        class="c4"
        id="medly-file-input"
        required=""
        type="file"
      />
    </label>
    <span
      class="c5 c6"
    >
      We will never share your email with anyone
    </span>
  </div>
</div>
`;

exports[`FileInput component should render properly with default props 1`] = `
.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-areas: 'label' 'field' 'description';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  width: 1px;
  opacity: 0;
}

.c1 {
  padding: 8px;
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  background-clip: padding-box;
  background-color: #ffffff;
  box-sizing: border-box;
  border: 1px solid #E0E0E0;
  border-radius: 5px;
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1:focus {
  border-color: #012040;
  outline: 0;
}

<div>
  <div
    class="c0"
    id="medly-file-input-field"
  >
    
    <label
      class="c1"
      id="medly-file-input-label"
    >
      Click to choose file
      <input
        class="c2"
        id="medly-file-input"
        type="file"
      />
    </label>
    
  </div>
</div>
`;
